lib/src/tools/analyze_tool.dart

```diff
@@ -0,0 +1,231 @@
+import 'dart:async';
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:args/command_runner.dart';
+import 'package:glob/glob.dart';
+import 'package:logging/logging.dart';
+
+import '../dart_dev_tool.dart';
+import '../utils/arg_results_utils.dart';
+import '../utils/assert_no_positional_args_nor_args_after_separator.dart';
+import '../utils/logging.dart';
+import '../utils/process_declaration.dart';
+import '../utils/run_process_and_ensure_exit.dart';
+
+final _log = Logger('Analyze');
+
+/// A dart_dev tool that runs the `dartanalyzer` on the current project.
+///
+/// To use this tool in your project, include it in the dart_dev config in
+/// `tool/dev.dart`:
+///     import 'package:dart_dev/dart_dev.dart';
+///
+///     final config = {
+///       'analyze': AnalyzeTool(),
+///     };
+///
+/// This will make it available via the `dart_dev` command-line app like so:
+///     pub run dart_dev analyze
+///
+/// This tool can be configured by modifying any of its fields:
+///     // tool/dev.dart
+///     import 'package:dart_dev/dart_dev.dart';
+///
+///     final config = {
+///       'analyze': AnalyzeTool()
+///         ..dartanalyzerArgs = ['--fatal-infos']
+///         ..include = [Glob('.'), Glob('other/**.dart')],
+///     };
+///
+/// It is also possible to run this tool directly in a dart script:
+///     AnalyzeTool().run();
+class AnalyzeTool extends DevTool {
+  /// The args to pass to the `dartanalyzer` process run by this tool.
+  ///
+  /// Run `dartanalyzer -h -v` to see all available args.
+  List<String> analyzerArgs;
+
+  @override
+  String description = 'Run static analysis on dart files in this package.';
+
+  /// The globs to include as entry points to run static analysis on.
+  ///
+  /// The default is `.` (e.g. `dartanalyzer .`) which runs analysis on all
+  /// public entry points in `lib/` and all dart files in other known
+  /// directories (`benchmark/`, `bin/`, `example/`, `test/`, `tool/`, `web/`),
+  /// as well as any dart files in the project root.
+  List<Glob> include;
+
+  @override
+  FutureOr<int> run([DevToolExecutionContext context]) =>
+      runProcessAndEnsureExit(buildProcess(context ?? DevToolExecutionContext(),
+          configuredAnalyzerArgs: analyzerArgs, include: include));
+
+  @override
+  Command<int> toCommand(String name) => DevToolCommand(name, this,
```

Comment:
Will this type parameter and the one on line 61 always be the same? If so, maybe the parameter should go on the class?

Reply:
I was trying to force the usage of an integer exit code via the `DevTool` class – it hardcodes the return type for `FutureOr<int> run()` and `Command<int> toCommand()`. Changing that to a parameterized type would defeat that purpose and allow `DevTool` implementations to return something other than an integer exit code.